_posts/2021-11-18-real-time-and-historical-ad.markdown

```diff
@@ -0,0 +1,68 @@
+
+Consider index `service_1_metrics` which contains different performance metrics for a service named `service_1`. The index `service_1_metrics` contains the fields `host`, `cpu_utilization`, `mem_utilization`, and `timestamp`, where `host` is the specific host that’s reporting the metrics, `cpu_utilization` is the current CPU utilization percentage reported by that host, `mem_utilization` is the current memory utilization percentage, and `timestamp` is the time when the metrics were collected. Now, an operator is interested in setting up anomaly detection on the CPU utilization and memory utilization metrics.
+
+In Step 1, users define their basic detector configuration - a detector is an individual anomaly task. You can have multiple detectors running in your cluster at any time. The detector settings include the detector name and description, details about the data source, and detector interval. Here, the operator would select the `service_1_metrics` index, and perhaps add some preprocessing to the data by adding a data filter to ignore any documents in the index where `host` is null. They would also select `timestamp` as the configured timestamp field. Under “Operation settings”, a user can tune how often they want to collect results with the “Detector interval” setting. The `service_1_metrics` collects results from all hosts every minute, and indexes a new document for each host every minute. In this case, using the default interval of 10 minutes is reasonable. which means the detector will aggregate the real-time source data and produce anomaly results every 10 minutes.
+
+![Define detector]({{ site.baseurl }}/assets/media/blog-images/2021-11-16-real-time-and-historical-ad/define-detector.png){: .img-fluid }
+
+In Step 2, the operator configures their features and other model settings. Here, the user may configure up to five features per detector. In this example, the operator creates two features - one for analyzing the CPU utilization metrics, and one for analyzing the memory utilization metrics. Under “Categorical fields”, the user may optionally select a field to partition the source data on. For example, if the operator selects the `host` field, a separate set of anomaly results will be produced for each unique host that is found in the index, allowing for a more fine-grained analysis of the data. By not selecting any categorical fields, all of the data will be treated as a single time series, and a single set of anomaly results will be produced.
```

Comment:
So there is an 'operator' and 'the user'- are these different individuals? It's unclear.